Emby.Server.Implementations/Data/SqliteUserRepository.cs

```diff
@@ -75,10 +75,12 @@ private void TryMigrateToLocalUsersTable(ManagedConnection connection)
             }
         }
 
-        private void RemoveEmptyPasswordHashes()
+        private void RemoveEmptyPasswordHashes(ManagedConnection connection)
         {
-            foreach (var user in RetrieveAllUsers())
+            foreach (var row in connection.Query("select id,guid,data from LocalUsersv2"))
```

Comment:
why change?

Reply:
`RetrieveAllUsers` does another `GetConnection` which will block indefinitely because we already got a connection.

Reply:
Maybe for another day so we can get this one delivered, but why ever are we waiting for WRITE lock before executing a read only query!?

Reply:
Can you move this code of getting the users inside `RetrieveAllUsers()` and make that function accept a `connection`? I don't like the amount of places where raw SQL is now...

Reply:
> 
> 
> Maybe for another day so we can get this one delivered, but why ever are we waiting for WRITE lock before executing a read only query!?

We only have a write connection, 🤷‍♂️ maybe a bit of a misnomer.